Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -853,9 +853,12 @@ void RenderElement::styleWillChange(StyleDifference diff, const RenderStyle& new
         bool willBeVisible = newStyle.visibility() == Visibility::Visible && !newStyle.skippedContentReason().has_value();
         if (wasVisible != willBeVisible) {
             if (CheckedPtr layer = enclosingLayer()) {
-                if (willBeVisible)
-                    layer->setHasVisibleContent();
-                else if (layer->hasVisibleContent() && (this == &layer->renderer() || layer->renderer().style().visibility() != Visibility::Visible))
+                if (willBeVisible) {
+                    if (m_style.skippedContentReason().has_value() && !newStyle.skippedContentReason().has_value())
```

Comment:
If `willBeVisible` is true, we know `!newStyle.skippedContentReason().has_value()` is true here.

Do we need to think about change in the other direction (unskipped to skipped)?

Reply:
Good point about not needing !newStyle.skippedContentReason().has_value() part of the check, removed.

I believe for unskipped to skipped, in the c-v: hidden case this is covered by the content-visibility-005.html test.

I think for c-v: auto it is less relevant since as soon as content is in the viewport it will count as unskipped, so I am not sure a "visual" test is possible. Let me know if you can think of a useful test scenario though.

Reply:
Actually I did come up today with two tests for the unskipped to skipped scenario while rebasing the patch, PTAL :)